projects/components/src/table/cells/state-parsers/table-cell-state-expander.ts

```diff
@@ -0,0 +1,17 @@
+import { StatefulTableRow, TableRowState } from '../../table-api';
+import { TableCellParser } from '../table-cell-parser';
+import { TableCellParserBase } from '../table-cell-parser-base';
+import { CoreTableCellParserType } from '../types/core-table-cell-parser-type';
+
+@TableCellParser({
+  type: CoreTableCellParserType.RowExpander
+})
+export class TableCellStateParserExpander extends TableCellParserBase<TableRowState, TableRowState, undefined> {
```

Comment:
Are these two parsers the same? Do we need to dupe it?

Reply:
They are same right now. I wanted to get feedback on whether I should make this return the whole object (more safer) or make it return the specific property.  I can update this PR depending on this.

Reply:
I'm fine with either:
- one copy of this
- no copies (and just use the no-op parser and update the renderers to read state off the row)

I think splitting to two specific ones is overkill - it's not _really_ safer because the renderer has the row anyway.

Reply:
Will move to one copy. Don't want to remove the parser.

Reply:
updated